pkg/kubectl/cmd/annotate.go

```diff
@@ -104,6 +104,7 @@ func NewCmdAnnotate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 		},
 	}
 	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDiscoveryFlags(cmd)
```

Comment:
Do you expect this discovery flag to be used for APIs other than ThirdPartyResources?